astropy/stats/funcs.py

```diff
@@ -1295,12 +1297,11 @@ def kuiper_false_positive_probability(D, N):
            and significance points", Biometrika, v.52, p.309, 1965.
 
     """
-    try:
-        from scipy.special import comb, factorial
-    except ImportError:
-        # Retained for backwards compatibility with older versions of scipy
-        # (factorial appears to have moved here in 0.14)
-        from scipy.misc import comb, factorial
+    if not HAS_SCIPY:
+        raise ModuleNotFoundError(
+            "scipy is required for kuiper_false_positive_probability"
```

Comment:
This changes the message from:

`ImportError: cannot import name 'comb' from 'scipy.<module>' `

to:

`scipy is required for kuiper_false_positive_probability`

I'll defer to @larrybradley whether this needs a change log or not.

Reply:
Technically, it is an API change, but I think it's an inconsequential one.  I doubt anyone is running this code and checking for the error type or message.

Reply:
OK I think everyone else think we do not need a change log, so let's go with the majority. Thanks for the feedback!